build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -90,32 +145,53 @@ impl Goodie for GraalVM {
 
 impl GraalVM {
     pub fn url(&self) -> BoxFuture<'static, Result<Url>> {
-        let platform_string = self.platform_string();
-        let graal_version_tag = self.graal_version.to_string_core();
-        let client = self.client.clone();
-        async move {
-            let repo = CE_BUILDS_REPOSITORY.handle(&client);
-            let release = repo.find_release_by_text(&graal_version_tag).await?;
-            crate::github::find_asset_url_by_text(&release, &platform_string).cloned()
+        match self.edition {
+            Edition::Community => {
+                let platform_string = self.platform_string();
+                let graal_version_tag = self.graal_version.to_string_core();
+                let client = self.client.clone();
+                async move {
+                    let repo = CE_BUILDS_REPOSITORY.handle(&client);
+                    let release = repo.find_release_by_text(&graal_version_tag).await?;
+                    crate::github::find_asset_url_by_text(&release, &platform_string).cloned()
+                }
+                .boxed()
+            },
+            Edition::Enterprise => {
+                let graal_version_tag = self.graal_version.to_string_core();
+                let url = format!(
+                    "https://download.oracle.com/graalvm/{}/archive/graalvm-jdk-{}_{}_bin.tar.gz",
+                    self.graal_version.major,
+                    graal_version_tag,
+                    self.os_arch_string(),
+                );
+                Box::pin(ready(
+                    Url::parse(&url).context("Failed to parse URL")
+                ))
+            }
         }
-        .boxed()
     }
 
-    pub fn platform_string(&self) -> String {
-        let Self { graal_version: _graal_version, arch, os, client: _client } = &self;
-        let os_name = match *os {
+    fn os_arch_string(&self) -> String {
+        let os_name = match self.os {
             OS::Linux => "linux",
             OS::Windows => "windows",
             OS::MacOS => "macos",
             other_os => unimplemented!("System `{}` is not supported!", other_os),
         };
-        let arch_name = match *arch {
+        let arch_name = match self.arch {
             Arch::X86_64 => "x64",
             Arch::AArch64 => "aarch64",
             other_arch => unimplemented!("Architecture `{}` is not supported!", other_arch),
         };
+        format!("{}-{}", os_name, arch_name)
+    }
+
+    pub fn platform_string(&self) -> String {
+        let Self { graal_version: _graal_version, client: _client , ..} = &self;
```

Comment:
I have mixed feelings about using `..` here. The pattern of destructuring `Self { fieds }` is usually used in places like this to force compile error when a new field id added — so the developer is forced to consider whether the new field should be part of the platform string or not.

(this is my thought rather than a change request)

Reply:
Destructuring pattern removed altogether in 10d41f87d5e806591bd18deb80915c5cf6ef5a34